tests/unit/ui/components/TestOverlay.js

```diff
@@ -1,73 +1,141 @@
 import * as React from 'react';
-import { renderIntoDocument, Simulate } from 'react-dom/test-utils';
 
-import Overlay from 'ui/components/Overlay';
+import {
+  applyUIStateChanges,
+  createFakeEvent,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import Overlay, { OverlayBase, extractId } from 'ui/components/Overlay';
 
 describe(__filename, () => {
-  function render(props = {}) {
-    return renderIntoDocument(<Overlay {...props} />);
-  }
+  const getProps = ({ ...props } = {}) => {
+    return {
+      id: 'Overlay',
+      className: 'Overlay',
+      store: dispatchClientMetadata().store,
+      ...props,
+    };
+  };
+
+  const render = ({ children, ...otherProps } = {}) => {
+    const html = '<div>a child div</div>';
+    return shallowUntilTarget(
+      <Overlay {...getProps(otherProps)}>{children || html}</Overlay>,
+      OverlayBase,
+    );
+  };
 
   it('renders an Overlay', () => {
     const root = render();
-    expect(root.overlayContainer).toBeTruthy();
-    expect(root.overlayBackground).toBeTruthy();
-    expect(root.overlayContainer.tagName).toEqual('DIV');
-    expect(root.overlayContainer.className).toContain('Overlay');
+    expect(root).toHaveClassName('Overlay');
+    expect(root.find('.Overlay-background')).toHaveLength(1);
+    expect(root.find('.Overlay-contents')).toHaveLength(1);
   });
 
   it('renders extra className if provided', () => {
-    const root = render({ className: 'I-am-so-over-it' });
-    expect(root.overlayContainer.className).toContain('Overlay');
-    expect(root.overlayContainer.className).toContain('I-am-so-over-it');
+    const className = 'I-am-so-over-it';
+    const root = render({ className });
+    expect(root).toHaveClassName(className);
   });
 
   it('renders children', () => {
-    const root = render({ children: 'hello' });
-    expect(root.overlayContents).toBeTruthy();
-    expect(root.overlayContents.textContent).toContain('hello');
+    const html = '<div>overriding child div</div>';
+    const root = render({ children: html });
+    expect(
+      root
+        .children()
+        .at(1)
+        .props().children,
```

Comment:
Is it possible to use the [children](http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html) wrapper functions instead?

Reply:
yep, let me take another look